presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -614,6 +655,39 @@ public void testTimestampWithTimeZone(boolean insertWithPresto)
         tests.execute(getQueryRunner(), dataSetup);
     }
 
+    @Test(dataProvider = "testTimestampWithTimeZoneDataProvider")
+    public void testArrayTimestampWithTimeZone(boolean insertWithPresto)
+    {
+        DataType<List<ZonedDateTime>> dataType;
+        DataSetup dataSetup;
+        if (insertWithPresto) {
+            dataType = arrayDataType(prestoTimestampWithTimeZoneDataType());
+            dataSetup = prestoCreateAsSelect("test_array_timestamp_with_time_zone");
+        }
+        else {
+            dataType = arrayDataType(postgreSqlTimestampWithTimeZoneDataType(), "timestamptz[]");
+            dataSetup = postgresCreateAndInsert("tpch.test_array_timestamp_with_time_zone");
+        }
+
+        DataTypeTest tests = DataTypeTest.create()
+                .addRoundTrip(dataType, asList(epoch.atZone(UTC), epoch.atZone(kathmandu)))
+                .addRoundTrip(dataType, asList(beforeEpoch.atZone(kathmandu), beforeEpoch.atZone(UTC)))
+                .addRoundTrip(dataType, asList(afterEpoch.atZone(UTC), afterEpoch.atZone(kathmandu)))
+                .addRoundTrip(dataType, asList(timeDoubledInJvmZone.atZone(UTC)));
+        if (!insertWithPresto) {
+            // Postgres results with non-DST time (winter time) for timeDoubledInJvmZone.atZone(jvmZone) while Presto results with DST time, so skip this test when insertWithPresto
+            // TODO: Possible fix:  columnValuesWithNames in io.prestosql.tests.datatype.CreateAsSelectDataSetup#literalInExplicitCast should contains timestamp values in utc. Same as with valueLiterals in io.prestosql.tests.datatype.CreateAndInsertDataSetup#insertRows
```

Comment:
I think it's not a good idea to normalize to UTC before writing, as this could hide some bugs (for example, the very bug you're considering here -- it _could_ be our fault)

Please remove this TODO.

Reply:
done